gaeguli/pipeline.c

```diff
@@ -648,6 +655,85 @@ _bus_watch (GstBus * bus, GstMessage * message, gpointer user_data)
   return G_SOURCE_CONTINUE;
 }
 
+static gboolean
+_is_target_device (GstDevice * target_device)
+{
+  gchar *device_class;
+  gboolean res = FALSE;
+
+  device_class = gst_device_get_device_class (target_device);
+
+  if (g_strrstr (device_class, PIPEWIRE_MEDIA_CLASS_STR)) {
+    res = TRUE;
+  }
+  g_free (device_class);
+
+  return res;
+}
+
+static guint
+_get_pw_nodeid (GstDevice * target_device, gchar * device_id)
+{
+  GstStructure *props;
+  guint node_id = 0;
+
+  props = gst_device_get_properties (target_device);
+  if (props) {
+    const gchar *node_str, *node_id_str;
+
+    if (gst_structure_has_field (props, GAEGULI_PIPELINE_TAG)
+        && gst_structure_has_field (props, PIPEWIRE_NODE_ID_STR)) {
+      node_str = gst_structure_get_string (props, GAEGULI_PIPELINE_TAG);
+      if (!g_strcmp0 (node_str, device_id)) {
+        node_id_str = gst_structure_get_string (props, PIPEWIRE_NODE_ID_STR);
+        node_id = atoi (node_id_str);
+        g_debug ("Got node_id = %d", node_id);
+      }
+    }
+    gst_structure_free (props);
+  }
+
+  return node_id;
+}
+
+static gchar *
+_get_pipeline_id_description (GaeguliPipeline * self)
```

Comment:
Could you maybe replace the duplicate parts of `_get_stream_props_description()` with a call of this function? These lines define how `gaeguli.pipeline_id` is put together and it's better to have them only at a single place in order to preserve consistency.